### PR TITLE
    mkl fast path patch update

### PR DESCRIPTION
**Pull request by glingyan**
1. enable fast path in nnvm branch
    2. fix mkl v2 private format problem
    3. disable mkl info

Signed-off-by: lingyan <lingyan.guo@intel.com>

### CODE REVIEW

File: src/operator/mkl/mkl_batch_norm-inl.h [hunk 1]
```diff
@@ -223,10 +232,11 @@ class MKLBatchNormOp : public Operator {
 #endif
       BatchNorm_res[dnnResourceDst] = fwd_top_data->prv_ptr();
     } else {
-      BatchNorm_res[dnnResourceDst] = reinterpret_cast<void *>(out.dptr_);
+      BatchNorm_res[dnnResourceDst] =
+        reinterpret_cast<void *>(out.dptr_);
     }
 
-    if (ctx.is_train && !param_.use_global_stats) {
```
Comment (piiswrong) [inline src/operator/mkl/mkl_batch_norm-inl.h]: why?

**Comment by piiswrong** [comment]
please remove the buffer in ndarray if possible

**Comment by glingyan** [comment]
Operator *CreateOp<cpu>(BatchNormParam param, int dtype) {
#if MXNET_USE_MKL2017 == 1
  if (!param.use_global_stats) {
    return new MKLBatchNormOp<cpu, float>(param);
  }

Do not need param_.use_global_stats beause we will fall back to mxnet implmenetation in batch_normal.cc

**Comment by glingyan** [comment]
I want to , first it to make it correct and then we found a better solution

**Comment by glingyan** [comment]
Hi Eric:
   This is the case that could not cover, it is only for "fwd_filter_data   @ MKLConvolutionOp"
  MXImperativeInvoke is call directly from c_api
int MXImperativeInvoke(AtomicSymbolCreator creator,
                       int num_inputs,
                       NDArrayHandle *inputs,
                       int *num_outputs,
                       NDArrayHandle **outputs,
                       int num_params,
                       const char **param_keys,
                       const char **param_vals)

libmxnet.dll!mxnet::MKLMemoryDescriptorBase<float>::convert_from_prv(void * cpu_ptr) Line 110        C++
              libmxnet.dll!mxnet::MKLMemHolder::check_and_prv_to_cpu(void * dptr_) Line 102               C++
    libmxnet.dll!mxnet::TBlob::FlatTo2D<mshadow::cpu,float>(mshadow::Stream<mshadow::cpu> * stream) Line 159            C++
              libmxnet.dll!mxnet::op::SGDMomUpdate<mshadow::cpu>(const nnvm::NodeAttrs & attrs, const mxnet::OpContext & ctx, const std::vector<mxnet::TBlob,std::allocator<mxnet::TBlob> > & inputs, const std::vector<enum mxnet::OpReqType,std::allocator<enum mxnet::OpReqType> > & req, const std::vector<mxnet::TBlob,std::allocator<mxnet::TBlob> > & outputs) Line 89           C++
              [External Code]
>             libmxnet.dll!MXImperativeInvoke::__l135::<lambda>(mxnet::RunContext rctx, mxnet::engine::CallbackOnComplete on_complete) Line 216          C++
              [External Code]
              libmxnet.dll!mxnet::engine::ThreadedEngine::ExecuteOprBlock(mxnet::RunContext run_ctx, mxnet::engine::OprBlock * opr_block) Line 327    C++
libmxnet.dll!mxnet::engine::ThreadedEnginePerDevice::CPUWorker<0>(mxnet::engine::ThreadedEnginePerDevice::ThreadWorkerBlock<0> * block) Line 181               C++
libmxnet.dll!mxnet::engine::ThreadedEnginePerDevice::PushToExecute::__l18::mxnet::engine::ThreadedEnginePerDevice::ThreadWorkerBlock<0> * <lambda>(void)::__l2::<lambda>() Line 77           C++
              [External Code]
              libmxnet.dll!invoke_thread_procedure(unsigned int(*)(void *) procedure, void * const context) Line 92 C++
              libmxnet.dll!thread_start<unsigned int (__cdecl*)(void * __ptr64)>(void * const parameter) Line 115        C++

**Comment by howard0su** [comment]
This patch generate  new warning:
src/operator/batch_norm.cc:48:1:   required from here
src/operator/./mkl/mkl_batch_norm-inl.h:146:18: error: unused variable ‘scale’ [-Werror=unused-variable]
     const real_t scale = static_cast<real_t>(in_data[batchnorm::kData].shape_[1]) /
                  ^
also, there is some spacing and code format issues like this line

    if (param_.fix_gamma)   slope = 1.f;

**Comment by glingyan** [comment]
will fix ,strange, not in my ubuntu

发自我的 iPhone

在 2016年12月30日，上午10:40，Howard0Su <notifications@github.com<mailto:notifications@github.com>> 写道：


This patch generate new warning:
src/operator/batch_norm.cc:48:1: required from here
src/operator/./mkl/mkl_batch_norm-inl.h:146:18: error: unused variable ‘scale’ [-Werror=unused-variable]
const real_t scale = static_cast<real_t>(in_data[batchnorm::kData].shape_[1]) /
^
also, there is some spacing and code format issues like this line

if (param_.fix_gamma)   slope = 1.f;


―
